log/log.go

```diff
@@ -47,6 +62,104 @@ func Trace(msg string, ctx ...interface{}) {
 	l.Output(msg, l.LvlTrace, CallDepth, ctx...)
 }
 
+func wrapCtx(sl SwapLogger, ctx ...interface{}) []interface{} {
+	for _, elem := range ctx {
+		if elem == "action" && len(ctx)%2 == 0 {
```

Comment:
This checks if the ctx is even, this avoids the error of unbalanced key value pairs

```
//ctx needs to be even because it's a series of key/value pairs
// no one wants to check for errors on logging functions,
// so instead of erroring on bad input, we'll just make sure
// that things are the right length and users can fix bugs
// when they see the output looks wrong
```

Reply:
is this not already checked by the `normalize` function though?

taking `Info` as an example, this would be reached through:

(`swap/log`) `Info` → (`go-ethereum/log/logger`) `Info` → `write` →  `newContext` → `normalize`

right?

Reply:
I will remove it, correct It is checked in normalize, but this adds the exception in the context.
I wanted to avoid polluting the log, but it will be the responsibility of the programmer to add the key pair accordingly.